### DISCUSSION TITLE
Guidelines - Line Length

### DISCUSSION THREAD

**Discussion by james-d-elliott** [authelia/authelia#3974] [General]
This is a discussion regarding potential line length guideline adjustments.

Context: https://github.com/authelia/authelia/pull/3950

**Comment by james-d-elliott** [comment]
@polarathene

I do appreciate the detailed information. This is probably why 80 chars is generally the recommended. Can I ask what resolution you're on?

I forgot to mention that it's also significantly easier to produce PR feedback, either via the GitHub tools, and via direct commentary. Long lines make it difficult to pinpoint the exact issue without significant additional detail.

As far as personal preference, the main areas of personal preference was extending the 80 chars to 120 as we felt the 2K resolution allowing side by side by side should reasonably allow 1K resolutions side by side.

**Reply by polarathene** [reply]
> I do appreciate the detailed information.

No worries, thank you for taking it into consideration :+1:

> Can I ask what resolution you're on?

I had been on 1080p until this year. I'm presently using a 1440p display, often with a window maximized or two side-by-side.

The gripe was at fixed width formatting for docs in general. I have no issue revising a paragraph and fiddling around with re-arranging the lines to required width, it's just tedious for content being rewritten.

I don't personally have any accessibility issues, but I have met of those that have - witnessing the trouble they go through to view content.

---

> I forgot to mention that it's also significantly easier to produce PR feedback, either via the GitHub tools, and via direct commentary.
> Long lines make it difficult to pinpoint the exact issue without significant additional detail.

Is that something you would expect to occur often when reviewing docs?

I implemented docs migration from Github Wiki to GH-Pages (+Netlify PR preveiws) for a popular project I am a maintainer of, and I cannot recall this ever being an issue with doc contributions.

Paragraphs shouldn't be massive to the point it's an issue when providing review feedback on a line. I'd rather the word-wrapped paragraph in full context, than a paragraph split across multiple lines and a review comment wedged into it to target a sentence / word in the middle of it. Easier for additional reviewers to read context when the whole paragraph is in scope above an existing review comment.

If there is ever a need to be more specific about the review feedback of that paragraph, it's a copy/paste with `>` for a quote. Little more effort maybe, but worth it.

My opinion on this is specifically for prose, not code. When I have to review actual logic, word-wrap and wide widths aren't helpful.

---

> As far as personal preference, the main areas of personal preference was extending the 80 chars to 120 as we felt the 2K resolution allowing side by side by side should reasonably allow 1K resolutions side by side.

That's fine. One of the maintainers I collaborate with often has a habit of using line breaks for code comments at 80 char width or less.  Something I find as too strict, and like you prefer to aim for around 100-120 chars as a rough boundary for line breaking at.

**Reply by james-d-elliott** [reply]
> No worries, thank you for taking it into consideration 👍

Not an issue at all. It's an area I want to eventually provide automated feedback in using markdownlint. Solidifying the guidelines now saves us having to reconsider or finding out that our reasoning is flawed later.

> Is that something you would expect to occur often when reviewing docs?
> 
> I implemented docs migration from Github Wiki to GH-Pages (+Netlify PR preveiws) for a popular project I am a maintainer of, and I cannot recall this ever being an issue with doc contributions.

> Paragraphs shouldn't be massive to the point it's an issue when providing review feedback on a line. I'd rather the word-wrapped paragraph in full context, than a paragraph split across multiple lines and a review comment wedged into it to target a sentence / word in the middle of it. Easier for additional reviewers to read context when the whole paragraph is in scope above an existing review comment.
> 
> If there is ever a need to be more specific about the review feedback of that paragraph, it's a copy/paste with `>` for a quote. Little more effort maybe, but worth it.

I think this is probably a personal preference thing. With word wrap, the GitHub review interface only allows you to select and suggest by line. However you can do multi-line selections and suggestions. This means it gives the reviewer the control on how specific an individual feedback item is. With word-wrapping you lose that control. Specifically if you want to provide two different suggestions for a single paragraph where either one should be committed then it's not possible. It also makes it easier to discuss a specific suggestion rather than addressing multiple suggestions in a single thread, though I think that's very rare to be an issue and even rarer in the docs themselves. 

As to if this is a particular issue for docs.. I'm not entirely sure if we've seen it a lot. I don't have any concrete data on it. I am fairy certain I've done it in the past but as we generally use the line length it's not as obvious as it would be if we didn't. 

> That's fine. One of the maintainers I collaborate with often has a habit of using line breaks for code comments at 80 char width or less. Something I find as too strict, and like you prefer to aim for around 100-120 chars as a rough boundary for line breaking at.

Yep, 80 is a *very common* guideline. I feel it's too restrictive with modern displays however understand why the 80 standard exists.

The guidelines around docs are not solidified yet. They may become so once we decide on a lint-able style guide as refactoring is a huge job. In our recent docs refresh which you commented on (the Pull Request link), I actually spent a lot of my time conforming them to a markdown lint. Interestingly the markdown line-length rule for most linters is on by default, enforced/strict, and has a length of 80.
